### PR TITLE
Extended RediSearch CI benchmarks to include mixed-read write benchmarks.

### PR DESCRIPTION
**Pull request by filipecosta90**
- Include mixed-read write benchmarks. 
- Included rate-limited benchmarks. 
- Extended the full-text variations.
- Split search related benchmarks into three concurrent groups to reduce time to have CI feedback.
- Updated union/interesection benchmarks (the benchmark input files were wrongly named )

### CODE REVIEW

**Comment by filipecosta90** [comment]
### Automated performance analysis summary

This comment was automatically generated given there is performance data available.

In summary:
- Detected a total of 9 stable tests between versions.

You can check a comparison in detail via the [grafana link](https://benchmarksrediscom.grafana.net/d//3Ejv2wZnk/?var-branch=master&var-branch=extend.benchmarks)

### Comparison between master and extend.benchmarks.

Time Period from 30 days ago. (environment used: oss-standalone)

|                          Test Case                           |Baseline master (median obs. +- std.dev)|Comparison extend.benchmarks (median obs. +- std.dev)|% change (higher-better)|                Note                |
|--------------------------------------------------------------|----------------------------------------|-----------------------------------------------------|------------------------|------------------------------------|
|ftsb-10K-enwiki_abstract-hashes-term-prefix                   | 8122 +- 1.0%  (7 datapoints)           | 8105 +- nan%  (1 datapoints)                        |-0.2%                   |-- no change --                     |
|ftsb-10K-enwiki_abstract-hashes-term-suffix                   | 2043 +- 0.8%  (7 datapoints)           | 2036 +- nan%  (1 datapoints)                        |-0.3%                   |-- no change --                     |
|ftsb-10K-enwiki_abstract-hashes-term-suffix-withsuffixtrie    | 78063 +- 5.6%  (7 datapoints)          | 74915 +- nan%  (1 datapoints)                       |-4.0%                   |waterline=5.6%. potential REGRESSION|
|ftsb-10K-enwiki_abstract-hashes-term-wildcard                 | 13370 +- 1.8%  (7 datapoints)          | 13233 +- nan%  (1 datapoints)                       |-1.0%                   |-- no change --                     |
|ftsb-10K-multivalue-numeric-json                              | 844 +- 1.6%  (7 datapoints)            | 861 +- nan%  (1 datapoints)                         |2.1%                    |-- no change --                     |
|ftsb-10K-singlevalue-numeric-json                             | 368 +- 1.1%  (7 datapoints)            | 368 +- nan%  (1 datapoints)                         |0.1%                    |-- no change --                     |
|ftsb-1K-enwiki_abstract-hashes-term-contains                  | 1756 +- 0.8%  (7 datapoints)           | 1747 +- nan%  (1 datapoints)                        |-0.5%                   |-- no change --                     |
|search-ftsb-10K-enwiki_abstract-hashes-term-withoutsuffix-trie| 36140 +- 2.4%  (7 datapoints)          | 36499 +- nan%  (1 datapoints)                       |1.0%                    |-- no change --                     |
|search-ftsb-10K-enwiki_abstract-hashes-term-withsuffix-trie   | 36389 +- 1.5%  (7 datapoints)          | 35633 +- nan%  (1 datapoints)                       |-2.1%                   |-- no change --                     |

File: tests/benchmarks/search-ftsb-10K-enwiki_pages-hashes-fulltext-mixed_simple-1word-query_write_1_to_read_20.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: "ftsb-10K-enwiki_pages-hashes-fulltext-mixed_simple-1word-query_write_1_to_read_20.yml"
+remote:
```
Comment (GuyAv46) [inline tests/benchmarks/search-ftsb-10K-enwiki_pages-hashes-fulltext-mixed_simple-1word-query_write_1_to_read_20.yml:2]:
```suggestion
name: "ftsb-10K-enwiki_pages-hashes-fulltext-mixed_simple-1word-query_write_1_to_read_20.yml"
description: "
             enwiki-abstract [details here](https://github.com/RediSearch/ftsb/blob/master/docs/enwiki-pages-benchmark/description.md), 
             from English-language Wikipedia:Database page edition data. 
             This use case generates 100K docs, with 3 TEXT fields (all sortable), 1 sortable TAG field, and 1 sortable NUMERIC fields per document.
             Specifically for this testcase:
                - Type (read/write/mixed): mixed
                - Query type: simple 1 word
                - Query sample: Lincoln
             "
remote:
```

**Review by GuyAv46 — COMMENTED** [review]
Thanks!
Are all the non-sortable scenarios include `SOERTBY` and `RETURN`? if so, I think we should mention it and maybe add an example of the command (not only for the query string)

**Review by GuyAv46 — COMMENTED** [review]
LGTM

**Review by GuyAv46 — DISMISSED** [review]

**Review by GuyAv46 — APPROVED** [review]

**Pull request merged**